### ISSUE TITLE
Logging breaks completely after JsonMappingException

### ISSUE THREAD

**Issue by angelo-v** [logstash/logstash-logback-encoder#253]
Try to log an object which causes a JsonMappingException -> All logging stops completly without notice.

**Example:**
```
public static void main(String[] args) {
        logger.info("line is logged successfully");
        logger.info("this fails to log without notice {}", keyValue("message", new Message("hello")));
        logger.info("even all further logs");
        logger.info("are not shown");
        logger.error("as well as errors", new RuntimeException("and logged exceptions"));
    }
}

public class Message {

    private String text;

    public Message(String text) {
        this.text = text;
    }

    @Override
    public String toString() {
        return text;
    }
}

```
logback.xml:
```

<?xml version="1.0" encoding="UTF-8"?>
<configuration>
    <appender name="CONSOLE" class="ch.qos.logback.core.ConsoleAppender">
        <encoder class="net.logstash.logback.encoder.LogstashEncoder">
            <provider class="net.logstash.logback.composite.loggingevent.ArgumentsJsonProvider"/>
        </encoder>
    </appender>
    <root level="INFO">
        <appender-ref ref="CONSOLE"/>
    </root>
</configuration>
```

**expected output:**
```
{"@timestamp":"2017-10-09T11:17:28.956+02:00","@version":1,"message":"line is logged successfully","logger_name":"Main","thread_name":"main","level":"INFO","level_value":20000}
{"@timestamp":"2017-10-09T11:17:28.974+02:00","@version":1,"message":"this fails to log without notice message=hello","logger_name":"Main","thread_name":"main","level":"INFO","level_value":20000,"message":{"text":"hello"}}
{"@timestamp":"2017-10-09T11:17:29.005+02:00","@version":1,"message":"even all further logs","logger_name":"Main","thread_name":"main","level":"INFO","level_value":20000}
{"@timestamp":"2017-10-09T11:17:29.005+02:00","@version":1,"message":"are not shown","logger_name":"Main","thread_name":"main","level":"INFO","level_value":20000}
{"@timestamp":"2017-10-09T11:17:29.007+02:00","@version":1,"message":"as well as errors","logger_name":"Main","thread_name":"main","level":"ERROR","level_value":40000,"stack_trace":"java.lang.RuntimeException: and logged exceptions\n\tat Main.main(Main.java:17)\n"}
```
**actual output:**
```
{"@timestamp":"2017-10-09T11:17:06.853+02:00","@version":1,"message":"line is logged successfully","logger_name":"Main","thread_name":"main","level":"INFO","level_value":20000}
```
Since `Message` has not Getter, Jackson fails. This failure is not communicated. Instead logging silently fails and all further log ouput of the application gets lost!

**Expected Behaviour**
The jackson exception should be logged. Further log output should not be affected.

You can find a fully working example of the erroneous behaviour here:
https://github.com/angelo-v/logstash-logback-encoder-bug

**Comment by philsttr** [comment]
Hi @angelo-v 

Can you try again with logstash-logback-encoder 4.11?

Logging errors are reported to [logback's status listener](https://logback.qos.ch/manual/configuration.html#automaticStatusPrinting).  Unfortunately, logback does not enable any status listeners by default.  You have to configure one yourself.

For example, I enabled status printing in your example project by adding the following line to logback.xml...

```xml
<statusListener class="ch.qos.logback.core.status.OnConsoleStatusListener" />
```

With logback-logstash-encoder 4.7 (as your example project references), the error is printed on the status listener, but further logging does not work...

```
$ java -cp jackson-annotations-2.6.0.jar:jackson-core-2.6.5.jar:jackson-databind-2.6.5.jar:logback-classic-1.1.7.jar:logback-core-1.1.7.jar:logstash-logback-encoder-4.7.jar:slf4j-api-1.7.22.jar:../build/libs/logstash-logback-encoder-bug-1.0-SNAPSHOT.jar Main
...
22:32:35,579 |-INFO in ch.qos.logback.core.joran.action.StatusListenerAction - Added status listener of type [ch.qos.logback.core.status.OnConsoleStatusListener]
22:32:35,592 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.core.ConsoleAppender]
22:32:35,596 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [CONSOLE]
22:32:35,738 |-INFO in ch.qos.logback.classic.joran.action.RootLoggerAction - Setting level of ROOT logger to INFO
22:32:35,738 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [CONSOLE] to Logger[ROOT]
22:32:35,738 |-INFO in ch.qos.logback.classic.joran.action.ConfigurationAction - End of configuration.
22:32:35,739 |-INFO in ch.qos.logback.classic.joran.JoranConfigurator@7a53c84a - Registering current configuration as safe fallback point
{"@timestamp":"2017-10-09T22:32:35.742-07:00","@version":1,"message":"line is logged successfully","logger_name":"Main","thread_name":"main","level":"INFO","level_value":20000,"HOSTNAME":"foo"}
{"@timestamp":"2017-10-09T22:32:35.760-07:00","@version":1,"message":"this fails to log without notice message=hello","logger_name":"Main","thread_name":"main","level":"INFO","level_value":20000,"HOSTNAME":"foo"22:32:35,771 |-ERROR in ch.qos.logback.core.ConsoleAppender[CONSOLE] - IO failure in appender com.fasterxml.jackson.databind.JsonMappingException: No serializer found for class Message and no properties discovered to create BeanSerializer (to avoid exception, disable SerializationFeature.FAIL_ON_EMPTY_BEANS) )
	at com.fasterxml.jackson.databind.JsonMappingException: No serializer found for class Message and no properties discovered to create BeanSerializer (to avoid exception, disable SerializationFeature.FAIL_ON_EMPTY_BEANS) )
	at 	at com.fasterxml.jackson.databind.ser.impl.UnknownSerializer.failForEmpty(UnknownSerializer.java:69)
	at 	at com.fasterxml.jackson.databind.ser.impl.UnknownSerializer.serialize(UnknownSerializer.java:32)
	at 	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:130)
	at 	at com.fasterxml.jackson.databind.ObjectMapper.writeValue(ObjectMapper.java:2383)
	at 	at com.fasterxml.jackson.core.base.GeneratorBase.writeObject(GeneratorBase.java:324)
	at 	at net.logstash.logback.marker.ObjectAppendingMarker.writeFieldValue(ObjectAppendingMarker.java:94)
	at 	at net.logstash.logback.marker.SingleFieldAppendingMarker.writeTo(SingleFieldAppendingMarker.java:86)
	at 	at net.logstash.logback.composite.loggingevent.ArgumentsJsonProvider.writeTo(ArgumentsJsonProvider.java:67)
	at 	at net.logstash.logback.composite.loggingevent.ArgumentsJsonProvider.writeTo(ArgumentsJsonProvider.java:41)
	at 	at net.logstash.logback.composite.JsonProviders.writeTo(JsonProviders.java:77)
	at 	at net.logstash.logback.composite.CompositeJsonFormatter.writeEventToGenerator(CompositeJsonFormatter.java:169)
	at 	at net.logstash.logback.composite.CompositeJsonFormatter.writeEventToOutputStream(CompositeJsonFormatter.java:146)
	at 	at net.logstash.logback.encoder.CompositeJsonEncoder.doEncode(CompositeJsonEncoder.java:73)
	at 	at net.logstash.logback.encoder.CompositeJsonEncoder.doEncode(CompositeJsonEncoder.java:30)
	at 	at ch.qos.logback.core.OutputStreamAppender.writeOut(OutputStreamAppender.java:187)
	at 	at ch.qos.logback.core.OutputStreamAppender.subAppend(OutputStreamAppender.java:212)
	at 	at ch.qos.logback.core.OutputStreamAppender.append(OutputStreamAppender.java:100)
	at 	at ch.qos.logback.core.UnsynchronizedAppenderBase.doAppend(UnsynchronizedAppenderBase.java:84)
	at 	at ch.qos.logback.core.spi.AppenderAttachableImpl.appendLoopOnAppenders(AppenderAttachableImpl.java:48)
	at 	at ch.qos.logback.classic.Logger.appendLoopOnAppenders(Logger.java:270)
	at 	at ch.qos.logback.classic.Logger.callAppenders(Logger.java:257)
	at 	at ch.qos.logback.classic.Logger.buildLoggingEventAndAppend(Logger.java:421)
	at 	at ch.qos.logback.classic.Logger.filterAndLog_1(Logger.java:398)
	at 	at ch.qos.logback.classic.Logger.info(Logger.java:583)
	at 	at Main.main(Main.java:14)
22:32:35,773 |-WARN in ch.qos.logback.core.ConsoleAppender[CONSOLE] - Attempted to append to non started appender [CONSOLE].
22:32:35,773 |-WARN in ch.qos.logback.core.ConsoleAppender[CONSOLE] - Attempted to append to non started appender [CONSOLE].
22:32:35,774 |-WARN in ch.qos.logback.core.ConsoleAppender[CONSOLE] - Attempted to append to non started appender [CONSOLE].
```

In logback-logstash-encoder 4.8, two changes were made to fix #190, which addresses the same problem you have reported.

* First, `SerializationFeature.FAIL_ON_EMPTY_BEANS` is now set to `false` by default, which will cause jackson serialization of your Message class to succeed.  So an error is no longer thrown in your example project.
* Second, if a different serialization error were to occur, the error will be logged to logback's status listener, but logging future events will continue to work.

```
$ java -cp jackson-annotations-2.6.0.jar:jackson-core-2.6.5.jar:jackson-databind-2.6.5.jar:logback-classic-1.2.3.jar:logback-core-1.2.3.jar:logstash-logback-encoder-4.11.jar:slf4j-api-1.7.22.jar:../build/libs/logstash-logback-encoder-bug-1.0-SNAPSHOT.jar Main
...
22:37:36,372 |-INFO in ch.qos.logback.core.joran.action.StatusListenerAction - Added status listener of type [ch.qos.logback.core.status.OnConsoleStatusListener]
22:37:36,383 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.core.ConsoleAppender]
22:37:36,388 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [CONSOLE]
22:37:36,525 |-INFO in ch.qos.logback.classic.joran.action.RootLoggerAction - Setting level of ROOT logger to INFO
22:37:36,526 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [CONSOLE] to Logger[ROOT]
22:37:36,526 |-INFO in ch.qos.logback.classic.joran.action.ConfigurationAction - End of configuration.
22:37:36,527 |-INFO in ch.qos.logback.classic.joran.JoranConfigurator@24158c28 - Registering current configuration as safe fallback point
{"@timestamp":"2017-10-09T22:37:36.528-07:00","@version":1,"message":"line is logged successfully","logger_name":"Main","thread_name":"main","level":"INFO","level_value":20000}
{"@timestamp":"2017-10-09T22:37:36.544-07:00","@version":1,"message":"this fails to log without notice message=hello","logger_name":"Main","thread_name":"main","level":"INFO","level_value":20000,"message":{}}
{"@timestamp":"2017-10-09T22:37:36.557-07:00","@version":1,"message":"even all further logs","logger_name":"Main","thread_name":"main","level":"INFO","level_value":20000}
{"@timestamp":"2017-10-09T22:37:36.557-07:00","@version":1,"message":"are not shown","logger_name":"Main","thread_name":"main","level":"INFO","level_value":20000}
{"@timestamp":"2017-10-09T22:37:36.558-07:00","@version":1,"message":"as well as errors","logger_name":"Main","thread_name":"main","level":"ERROR","level_value":40000,"stack_trace":"java.lang.RuntimeException: and logged exceptions\n\tat Main.main(Main.java:17)\n"}
```

**Comment by angelo-v** [comment]
Hi @philsttr 
thanks for your quick and in-depth reply. Version 4.11 works fine! Should have checked that first, sorry!
